### PR TITLE
Support the "nonce" parameter forwarding without a session

### PR DESCRIPTION
**Pull request by skycocker**
I'll rebase the README with the updates from https://github.com/omniauth/omniauth_openid_connect/pull/128 as soon as that gets the necessary second review.

### CODE REVIEW

File: lib/omniauth/strategies/openid_connect.rb [hunk 1]
```diff
@@ -353,7 +353,7 @@ def verify_id_token!(id_token)
 
         decode_id_token(id_token).verify!(issuer: options.issuer,
                                           client_id: client_options.identifier,
-                                          nonce: stored_nonce)
+                                          nonce: params['nonce'] || stored_nonce)
```
[the remark this answers is not in the document]
Comment (skycocker) [reply lib/omniauth/strategies/openid_connect.rb]: Good catch - we probably do. Updated.

File: lib/omniauth/strategies/openid_connect.rb [hunk 2]
```diff
@@ -378,7 +378,7 @@ def verify_id_token!(id_token)
 
         decode_id_token(id_token).verify!(issuer: options.issuer,
                                           client_id: client_options.identifier,
-                                          nonce: stored_nonce)
+                                          nonce: params['nonce'].presence || stored_nonce)
```
Comment (stanhu) [inline lib/omniauth/strategies/openid_connect.rb:381]: Isn't presence only a Rails-specific method? I'm not sure how the test case below passed.

```ruby
% irb
irb(main):001:0> foo = "hi"
=> "hi"
irb(main):002:0> foo.presence
(irb):2:in `<main>': undefined method `presence' for "hi":String (NoMethodError)
Did you mean?  prepend
	from /Users/stanhu/.asdf/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
	from /Users/stanhu/.asdf/installs/ruby/3.1.2/bin/irb:25:in `load'
	from /Users/stanhu/.asdf/installs/ruby/3.1.2/bin/irb:25:in `<main>'
```

File: test/lib/omniauth/strategies/openid_connect_test.rb [hunk 3]
```diff
@@ -237,6 +237,38 @@ def test_callback_phase_with_id_token
         strategy.callback_phase
       end
 
+      def test_callback_phase_with_id_token_and_param_provided_nonce # rubocop:disable Metrics/AbcSize
+        code = SecureRandom.hex(16)
```
Comment (stanhu) [inline test/lib/omniauth/strategies/openid_connect_test.rb:241]: `code` and `nonce` can probably be dropped due to #132.

File: lib/omniauth/strategies/openid_connect.rb (hunk 2 above)
Comment (skycocker) [reply lib/omniauth/strategies/openid_connect.rb:381]: You're right, it's the second time I'm trying to use rails helper methods in this repo. Anyway, the sad part is that it doesn't seem like any of the test examples in the whole gem test all that much 😅 it's all stubbed and doesn't reach that code. I'll try to refactor this as part of this PR when I have a minute.

Comment (stanhu) [reply lib/omniauth/strategies/openid_connect.rb:381]: Interesting, `activemodel` is a dependency of `openid_connect`, so as a result `activesupport` is available.

**Pull request merged**